### ISSUE TITLE
Horizontal spacing between equations of fractions

### ISSUE THREAD

**Issue by Cirran** [Khan/KaTeX#312]
Hi! I'm not sure it looks right. Please, check it. Simple demo.
![bug](https://cloud.githubusercontent.com/assets/3097333/8743492/12fa34ba-2c78-11e5-9d24-2b5452cd1fc6.PNG)
```html
<!doctype html>
<html lang="ru">
  <head>
    <meta charset="UTF-8">
    <title>Bug of rendering?</title>
    <script src="katex/katex.min.js" type="text/javascript"></script>
    <link href="katex/katex.min.css" rel="stylesheet" type="text/css">
    <script src="katex/auto-render.min.js" type="text/javascript"></script>
    <style type="text/css">
      body {
          margin: 0px;
          padding: 0px;
          font-size: 50px;
      }
    </style>
  </head>
  <body>
    A) $3 * \dfrac{40}{50}$<br />
    B) $\dfrac{40}{50} * \dfrac{39}{50} * \dfrac{38}{50}$<br />
    C) $\dfrac{40}{50} * \dfrac{40}{50} * \dfrac{40}{50}$<br />
    D) $\dfrac{40}{50} * \dfrac{39}{49} * \dfrac{38}{48}$<br />
    <script>
      renderMathInElement(
          document.body,
          {
              delimiters: [
                  {left: "$", right: "$", display: false}
              ]
          }
      );
    </script>
  </body>
</html>
```

**Comment by PeterBocan** [comment]
try: display: true

**Comment by Cirran** [comment]
Sorry, I don't need displayed equations. For them I can add `{left: "$$", right: "$$", display: true}` or use `\displaystyle` command. I need only inline equations.

**Comment by kevinb7** [comment]
@Cirran not sure when I'll have time to look at this, but in the meantime you could improve the spacing by putting each line inside a block element like a <p> or <div> or use a list with `list-style-type:none;' and adjust the padding/margin as necessary.

**Comment by Cirran** [comment]
@kevinb7 Thank you, the wrapping them into block `<div style="line-height: 2.75;"> ... </div>` has helped me. Sorry for troubling.

**Comment by kevinb7** [comment]
@Cirran no worries.  I'm going to leave this issue open because we should, at some point, try to find a better solution to this problem.

**Comment by xymostech** [comment]
Tinkering around with things, I think this might be caused by a couple problems:

1. We make our fonts a little bit thicker when we create them, so we maybe should adjust the metrics that we get to account for this?
2. TeX assumes that you're going to put some space between the different lines, but browsers don't always? Maybe we should explicitly add the line-height ourselves?

**Comment by spicyj** [comment]
Yeah, ideally we can adjust the line heights so that this looks right. We should be careful to not increase the line height for inline (non-fraction) math in regular text though, otherwise you get uneven spacing when using math on only a few lines of a large paragraph.

**Comment by mbourne** [comment]
Hi

On a related issue, the vertical spacing when using fractions in matrices is also too tight.

![image](https://cloud.githubusercontent.com/assets/7541940/8949305/5c316e80-35e4-11e5-9beb-77d336b02415.png)

![image](https://cloud.githubusercontent.com/assets/7541940/8949331/a2a16744-35e4-11e5-8c32-67024bb581f8.png)

How do we space them out some more?

Regards
Murray

**Comment by mbourne** [comment]
Any progress on this issue?

**Comment by kevinb7** [comment]
@mbourne I think what you're seeing is a separate issue.  Can you create a separate issue and include which browser/os combo you're seeing this in.  Sorry for not saying something sooner.

**Comment by kevinbarabash** [comment]
@edemaine I wonder if we need a `jot` between rows in `*matrix` environments?

**Comment by edemaine** [comment]
@kevinbarabash @mbourne In fact, that's how it's supposed to render, using LaTeX as a reference.  It renders the first matrix as follows:

![pmatrix](https://user-images.githubusercontent.com/2218736/27006610-51f09356-4e06-11e7-880b-ba3a1c72f9a7.png)

You can improve the situation, in both LaTeX and KaTeX, by providing an optional argument to `\\`.  For example:

```latex
=\begin{pmatrix}
    \frac{1}{5} & \frac{5}{12} & \frac{7}{20} \\[0.5em]
    \frac{1}{5} & \frac{1}{6} & \frac{1}{10} \\[0.5em]
    \frac{1}{5} & -\frac{1}{6} & \frac{1}{10}
  \end{pmatrix}
```

**Comment by sophiebits** [comment]
I'm not convinced that it makes sense to adjust the metrics for the "blacker" fonts – IIRC this lever was originally designed so that different printers could print Computer Modern in a way that looked good, even if an individual printer might end up printing the lines a little thicker or less thick. Similarly, different screen rasterization algorithms might do the same. But I think that's different from the underlying font metrics – if you were to use a bold face then you would typically have wider metrics but the goal with thickening fonts artificially is that after production they look the correct (original) thickness and therefore you wouldn't want to adjust the distance between the characters.

**Comment by kevinbarabash** [comment]
I think the spacing issue will get better once we support `\setlength` see https://github.com/Khan/KaTeX/issues/687.

**Comment by q2apro** [comment]
I ran into the linespacing with problems too, in various Latex code.

The proposed line-height solution was not good because it created a padding on top of each Katex-Block.

Instead I found another solution by CSS:

```
.katex .katex-html>.newline {
    display: block;
    padding-bottom: 10px;
}
```

Hope that helps.
